### PR TITLE
Issue #824 Track services errors as messages in Sentry

### PR DESCRIPTION
**Pull request by bensinjin**
This PR allows us to be notified when error screens are displayed from the services screen. Sentry does a good job of collecting other various metadata like for instance OS so we don't need to worry about sending that data along in the message. Here's a screenshot of what this looks like in my test Sentry instance:

Screen Shot 2019-10-30 at 9 58 55 AM

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import { Application } from './lib/application';
-import Sentry from 'sentry-expo';
+import * as Sentry from 'sentry-expo';
 
-// DSN for PeaceGeek's pathways project.
-Sentry.config('https://de23d08a2e2b44a49849c045c2a6fc0c@sentry.io/256284').install();
+Sentry.init({
+    // DSN for PeaceGeek's pathways project
+    dsn: 'https://de23d08a2e2b44a49849c045c2a6fc0c@sentry.io/256284',
```
Comment (rasmus-storjohann-PG) [inline App.js]: Should this be checked in? I think all the values in this config, including debug, should be defined in .env

File: src/components/error_screens/ErrorScreenSwitcherComponent.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// tslint:disable:no-expression-statement
```
Comment (rasmus-storjohann-PG) [inline src/components/error_screens/ErrorScreenSwitcherComponent.tsx]: Why is this change needed?

File: src/components/services/service_list_component.tsx [hunk 3]
```diff
@@ -18,6 +19,9 @@ import { ErrorScreenSwitcherComponent } from '../error_screens/ErrorScreenSwitch
 import { Errors } from '../../validation/errors/types';
 import { EmptyListComponent } from '../empty_component/empty_list_component';
 import { LatLong } from '../../validation/search/types';
+import { getSentryMessageForError } from '../../validation/errors/sentry_messages';
+
+const SENTRY_ERROR_CONTEXT = 'ServicesListing';
```
Comment (rasmus-storjohann-PG) [inline src/components/services/service_list_component.tsx]: We have a `constants` file for this.

File: src/components/services/service_list_component.tsx [hunk 4]
```diff
@@ -53,10 +59,12 @@ export const ServiceListComponent = (props: Props): JSX.Element => {
...
+        const errorType = props.topicServicesOrError.errorMessageType;
+        Sentry.captureMessage(getSentryMessageForError(errorType, SENTRY_ERROR_CONTEXT));
         return (
             <ErrorScreenSwitcherComponent
                 refreshScreen={refreshServicesData}
-                errorType={props.topicServicesOrError.errorMessageType}
+                errorType={errorType}
                 header={
                     <ServiceListHeaderComponent title={props.topic.title} />
                 }
```
Comment (rasmus-storjohann-PG) [inline src/components/services/service_list_component.tsx]: There is a lot of code duplication here, you can DRY this up I think.

File: App.js (hunk 1 above)
Comment (bensinjin) [reply App.js]: Fixed

File: src/components/error_screens/ErrorScreenSwitcherComponent.tsx (hunk 2 above)
Comment (bensinjin) [reply src/components/error_screens/ErrorScreenSwitcherComponent.tsx]: Removed

File: src/components/services/service_list_component.tsx (hunk 3 above)
Comment (bensinjin) [reply src/components/services/service_list_component.tsx]: Fixed

File: src/components/services/service_list_component.tsx (hunk 4 above)
Comment (bensinjin) [reply src/components/services/service_list_component.tsx]: Fixed
